packages/realm-server/migrations/1728561353454_create-user-billing.js

```diff
@@ -0,0 +1,164 @@
+exports.up = (pgm) => {
+  pgm.createTable('users', {
+    id: {
+      type: 'serial',
+      primaryKey: true,
+    },
+    matrix_user_id: { type: 'varchar', notNull: true, unique: true },
+    stripe_customer_id: { type: 'varchar' },
+    created_at: {
+      type: 'timestamp',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+  });
+
+  pgm.createTable('plans', {
+    id: {
+      type: 'serial',
+      primaryKey: true,
+    },
+    name: { type: 'varchar', notNull: true },
+    monthly_price: { type: 'numeric(19,4)', notNull: true },
+    credits_included: { type: 'integer', notNull: true },
+    overage_fee_per_credit_in_usd: { type: 'numeric(19,4)' },
+    created_at: {
+      type: 'timestamp',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+  });
+
+  pgm.createTable('billing_cycles', {
```

Comment:
The idea of this table is that when Stripe sends a webhook that charge has been made, we will save the period start and end so that we will be able to calculate the credits used in this period (it will give us the answer to the question "how many credits has this user used in the current billing cycle")

Reply:
In my opinion, it would be better to store the total credits used somewhere, so we don't have to recalculate it each time we fetch the data.

Reply:
The aggregation should be really fast and simple (just a sum over records in a certain period) and makes sure the data is correct. Caching of used credits would need extra logic, for a little bit more convenience and bigger chance of an outdated aggregation. Not sure if it's worth it at this point where loads aren't that big.

Reply:
@FadhlanR I actually disagree. we want an append only table so that we can have an accurate historical record. this will be important for auditing. In the future if we want to have an easy way to calculate this we could create a view on this table.